### PR TITLE
Extract npm 7 project fixtures and update specs

### PR DESCRIPTION
**Pull request by feelepxyz**
Extract all project based spec fixtures from npm7 pr to reduce diff:
https://github.com/dependabot/dependabot-core/pull/3011

The file updater specs changes are copied but excluding the npm 7
specifc specs.

### CODE REVIEW

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 1]
```diff
@@ -1138,57 +1138,6 @@
       end
     end
 
-    context "when updating a sub dependency with multiple requirements" do
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1141]: Moved down to npm 6 specific tests

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 2]
```diff
@@ -1138,57 +1138,6 @@
...
-      let(:files) { project_dependency_files("yarn/multiple_sub_dependencies") }
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1142]: Moved down to yarn specific tests

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 3]
```diff
@@ -1138,57 +1138,6 @@
...
-
-      it "de-duplicates all entries to the same version" do
-        expect(updated_files.map(&:name)).to match_array(["yarn.lock"])
-        expect(updated_yarn_lock.content).
-          to include("js-yaml@^3.10.0, js-yaml@^3.4.6, js-yaml@^3.9.0:\n"\
-                     '  version "3.12.0"')
-      end
-    end
-
-    context "with a .npmrc" do
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1158]: Moved down to npm 6 specific tests

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 4]
```diff
@@ -1214,178 +1163,6 @@
...
-        [{
-          file: "package.json",
-          requirement: "^2.0.1",
-          groups: ["dependencies"],
-          source: nil
-        }]
-      end
-
-      context "with an npm lock file" do
-        let(:files) { project_dependency_files("npm6/latest_package_requirement") }
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1239]: Moved down to npm 6 specific tests

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 5]
```diff
@@ -1214,178 +1163,6 @@
...
-            to match_array(%w(package.json package-lock.json))
-          expect(updated_npm_lock.content).
-            to include("extend/-/extend-3.0.2.tgz")
-          expect(updated_npm_lock.content).
-            to include("etag/-/etag-1.7.0.tgz")
-        end
-      end
-
-      context "with a yarn lockfile" do
-        let(:files) { project_dependency_files("yarn/latest_package_requirement") }
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1252]: Moved down to yarn specific tests

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 6]
```diff
@@ -1214,178 +1163,6 @@
...
-          expect(updated_yarn_lock.content).
-            to include("extend@^3.0.2:\n  version \"3.0.2\"")
-          expect(updated_yarn_lock.content).
-            to include("etag@latest:\n  version \"1.7.0\"")
-        end
-      end
-    end
-
-    context "when the exact version we're updating from is still requested" do
-      let(:files) { project_dependency_files("yarn/typedoc-plugin-ui-router") }
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1266]: Moved down to yarn specific tests

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 7]
```diff
@@ -1214,178 +1163,6 @@
...
-
-        expect(updated_yarn_lock.content).
-          to include("typescript@2.1.4:\n  version \"2.1.4\"")
-        expect(updated_yarn_lock.content).
-          to include("typescript@^2.1.1:\n  version \"2.9.1\"")
-      end
-    end
-
-    describe "the updated package-lock.json" do
-      let(:files) { project_dependency_files("npm6/simple") }
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1292]: Moved down to npm 6 specific tests

File: npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb [hunk 8]
```diff
@@ -1214,178 +1163,6 @@
...
-          expect(
-            parsed_lockfile.dig(
-              "dependencies", "fetch-factory", "requires", "es6-promise"
-            )
-          ).to eq("^3.0.2")
-        end
-      end
-    end
-
-    describe "the updated yarn_lock" do
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb:1317]: Moved down to yarn specific tests

**Review by jurre — APPROVED** [review]

**Pull request merged**